### PR TITLE
[4.0] Generate the classmap from the manifest files

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #20823.

### Summary of Changes
Creates the namespace map directly from the manifest files instead of the database. This solves the issue that we are now able to declare a folder where the root of the namespace is. In the future it can be extended to even support multiple namespaces, etc. There is also no database involved here.

Regarding the performance impact. It takes for the core extensions on a vagrant box with shared folders and no performance optimization 29 milliseconds to execute.

### Testing Instructions
- Delete the file libraries/autoload_psr4.php.
- Open the Joomla front end.

### Expected result
All is working as expected.

### Actual result
All is working as expected.

### CODE REVIEW

**Comment by laoneo** [comment]
If there is no more feedback here, I can fix the merge conflicts and get this merged.

**Comment by wilsonge** [comment]
Go for it. We can always put it back into the db if performance comes an issue closer to release date. And this is simpler

**Comment by laoneo** [comment]
Ok. Your turn.

File: libraries/namespacemap.php [hunk 1]
```diff
@@ -69,57 +62,16 @@ public function ensureMapFileExists()
 	 */
 	public function create()
 	{
-		$extensions = $this->getNamespacedExtensions();
-
-		$elements = array();
+		$extensions = $this->getNamespaces('administrator/components');
```
Comment (wilsonge) [inline libraries/namespacemap.php]: Just doing this on code review but aren't you missing the frontend components here?

**Comment by wilsonge** [comment]
Turn complete :) thanks!

**Pull request merged**